src/main/java/com/iotiq/application/service/CustomerService.java

```diff
@@ -115,12 +120,18 @@ public void createCustomer(User currentUser) {
         customer.setMedicalData(new MedicalData());
         customer.setSizeInfo(new SizeInfo());
         customerRepository.save(customer);
+        log.info("User's {} customer created", currentUser.getId());
     }
 
     private List<OrderDto> getLastTwoOrders(List<Order> orderList) {
         if (CollectionUtils.isEmpty(orderList)) {
             return Collections.emptyList();
         }
-        return orderList.stream().map(order -> ModelMapperUtil.map(order, OrderDto.class)).sorted(Comparator.comparing(OrderDto::getOrderDate).reversed()).limit(2).collect(Collectors.toList());
+        return orderList.stream()
+                .map(order -> order == null ? null : ModelMapperUtil.map(order, OrderDto.class))
```

Comment:
you can first do a filter to filter the nonnull and visible orders only, then sort and limit so that you don't map unnecessary orders. 
Or the better option is to write a repository function to fetch only the last two orders :) (This may be a future enhancement)